### PR TITLE
Added clear and all operations

### PR DESCRIPTION
**Pull request by MaanooAk**
Clear (per #9 request) and get all points operations

### CODE REVIEW

**Comment by kleinesfilmroellchen** [comment]
The clear function must also call `clear` recursively on all sub-quadtrees!

File: quadtree.js [hunk 1]
```diff
@@ -90,6 +90,17 @@ class QuadTree {
     this.divided = false;
   }
 
+  clear() {
+    if (this.divided) {
+      delete this.northeast;
+      delete this.northwest;
+      delete this.southeast;
+      delete this.southwest;
```
[the remark this answers is not in the document]
Comment (MaanooAk) [reply quadtree.js]: When the QuadTree is created, it doesn't have the 4 properties (make sense, less memory), so when the clear is called the result must be the same as if we just created a new one to be consistent.

File: quadtree.js [hunk 2]
```diff
@@ -90,6 +90,17 @@ class QuadTree {
     this.divided = false;
   }
 
+  clear() {
+    if (this.divided) {
+      delete this.northeast;
```
[the remark this answers is not in the document]
Comment (MaanooAk) [reply quadtree.js]: No, the gc with take care of them. Calling the clear recursively will do nothing (just slow it down)

**Comment by MaanooAk** [comment]
@kleinesfilmroellchen There is no need, the gc with take care of them. Calling the clear recursively will do nothing (just slow it down)

File: quadtree.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kleinesfilmroellchen) [reply quadtree.js]: As you might know, JavaScript's GC is not operating that often. I think this should be done cleanly by controlling object existence explicitly and not letting an application whose behavior is not very well known do the job. The heap might get out of hand if you do a lot of quadtree stuff in short time.

[the remark this answers is not in the document]
Comment (MaanooAk) [reply quadtree.js]: Even if you call the clear to the sub-QuadTrees they will still be on the heap waiting for the gc. Even with the delete keyword, you still wait for the gc. As far as I know you cant control the object existence explicitly

[the remark this answers is not in the document]
Comment (kleinesfilmroellchen) [reply quadtree.js]: Well, I don't know much about JavaScript GC, but to me, it feels safer marking an object explicitly as "safe to delete". Someone might consider running a test on this to see performance times under different circumstances.

File: quadtree.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pierstoval) [reply quadtree.js]: Yep, your PoV about consistency is legit, but I even think that these properties should be nulled in the beginning, after a construct :p

**Comment by MaanooAk** [comment]
Resolved the conflicts.

**Pull request closed without merging**